iothub_client/samples/iothub_client_device_cbor_sample/iothub_client_device_cbor_sample.c

```diff
@@ -0,0 +1,956 @@
+    {
+        (void)printf("Failed to create device client from connection string.\n");
+        exit(EXIT_FAILURE);
+    }
+
+    //
+    // Set Options
+    //
+    bool trace_on = true; // Debugging
+    rc = IoTHubDeviceClient_SetOption(iothub_client_handle, OPTION_LOG_TRACE, &trace_on);
+    if (rc != IOTHUB_CLIENT_OK)
+    {
+        (void)printf("Failed to set option for %s: return code %d.\n", OPTION_LOG_TRACE, rc);
+        exit(rc);
+    }
+
+    // Set the auto URL Encoder (recommended for MQTT). Please use this option unless you are URL
+    // Encoding inputs yourself. ONLY valid for use with MQTT.
+    bool url_encode_on = true;
+    rc = IoTHubDeviceClient_SetOption(iothub_client_handle, OPTION_AUTO_URL_ENCODE_DECODE, &url_encode_on);
+    if (rc != IOTHUB_CLIENT_OK)
+    {
+        (void)printf("Failed to set option for %s: return code %d.\n", OPTION_AUTO_URL_ENCODE_DECODE, rc);
+        exit(rc);
+    }
+
+    // Format ONLY the device twin document using CBOR.
+    // ONLY valid for use with MQTT. Must occur prior to CONNECT.
+    OPTION_TWIN_CONTENT_TYPE_VALUE ct = OPTION_TWIN_CONTENT_TYPE_CBOR;
+    rc = IoTHubDeviceClient_SetOption(iothub_client_handle, OPTION_TWIN_CONTENT_TYPE, &ct);
+    if (rc != IOTHUB_CLIENT_OK)
+    {
+        (void)printf("Failed to set option for %s: return code %d.\n", OPTION_TWIN_CONTENT_TYPE, rc);
+        exit(rc);
+    }
+
+#ifdef SET_TRUSTED_CERT_IN_SAMPLES
+    // Use SDK-supplied trusted certificates to run the sample.
+    // ONLY to be used for the sample. **NOT to be used in production code.**
+    rc = IoTHubDeviceClient_SetOption(iothub_client_handle, OPTION_TRUSTED_CERT, certificates);
+    if (rc != IOTHUB_CLIENT_OK)
+    {
+        (void)printf("Failed to set option for %s: return code %d.\n", OPTION_TRUSTED_CERT, rc);
+        exit(rc);
+    }
+#endif
+}
+
+static void connect_device_client_send_and_receive_messages(void)
+{
+    int rc;
+
+    //
+    // Send and receive messages from IoT Hub. Connection occurs when a message is first sent.
+    //
+
+    // Set callback for connection status to IoT Hub.
+    rc = IoTHubDeviceClient_SetConnectionStatusCallback(iothub_client_handle, connection_status_callback, NULL);
+    if (rc != IOTHUB_CLIENT_OK)
+    {
+        (void)printf("Failed to set the connection status callback: return code %d.\n", rc);
+        exit(rc);
+    }
+
+    // Set callback for GET twin document request. - Approach #1 to requesting the twin document.
+    // Send asynchronous GET request for twin document. Connection will occur here.
+    rc = IoTHubDeviceClient_GetTwinAsync(iothub_client_handle, get_twin_async_callback, NULL);
+    if (rc != IOTHUB_CLIENT_OK)
+    {
+        (void)printf("Failed to send a GET request for the twin document asynchronously: return code %d.\n", rc);
+        exit(rc);
+    }
+
+    // Set callback for twin desired property PATCH messages from IoT Hub. - Approch #2 to requesting the twin document.
```

Comment:
nit: Consider not calling this an "approach" to getting the twin document. Getting it I think is a side effect that maybe is a debatable design decision at this point. I think leaving it for its self-named purpose offers less confusion for customers and leaves the `_GetTwinAsync` as the only option for consciously getting the twin doc.